src/connector/src/source/pulsar/source/reader.rs

```diff
@@ -174,8 +176,12 @@ impl SplitReader for PulsarBrokerReader {
             .with_topic(&topic)
             .with_subscription_type(SubType::Exclusive)
             .with_subscription(format!(
-                "rw-consumer-{}-{}",
-                source_ctx.fragment_id, source_ctx.actor_id
+                "{}-{}-{}",
+                props
+                    .subscription_name_prefix
+                    .unwrap_or(PULSAR_DEFAULT_SUBSCRIPTION_PREFIX.to_string()),
+                source_ctx.fragment_id,
+                source_ctx.actor_id
```

Comment:
Is actor_id needed?

Reply:
we use exclusive subscription here, which mean a partition can only be consumed by one subscription. So actor-id is essential to avoid collision.

Reply:
1 pulsar split corresponds to 1 topic. In one source (i.e., one fragment), each actor will be assigned to different topics. So actor_id is not necessary. There's no collision.

Reply:
> 1 pulsar split corresponds to 1 topic. In one source (i.e., one fragment), each actor will be assigned to different topics.

I dont get the logic

Reply:
The assignment looks like this. Where's the collision? Please give an example

- fragment 1
  * actor 1001  - `topic-partition-1`
  * actor 1002  - `topic-partition-2`
  * actor 1003  - `topic-partition-3`
  * ...

Reply:
I am afraid things not work in this way

> The exclusive type is a subscription type that only allows a single consumer to attach to the subscription. If multiple consumers subscribe to a topic using the same subscription, an error occurs. Note that if the topic is partitioned, all partitions will be consumed by the single consumer allowed to be connected to the subscription. [source](https://pulsar.apache.org/docs/3.3.x/concepts-messaging/#exclusive)

We cannot attach multiple consumers to one subscription in `exclusive` mode. Since we want to manage the data parallelism, ie. partitions, in RisingWave, we have to separate the consumers from each parallelism into different subscriptions.

Reply:
I believe each subscription belongs to a different (sub)topic... They are not "one subscription"

Reply:
In the example above, `rw-consumer-6` are 3 different subscriptions

Reply:
We already  separated the consumers by subscribing to different sub topics. It's not by separate them into different subscriptions..